### PR TITLE
Add modulo operator to the query language

### LINKED ISSUE #7553: % operator
Proposal: Would like the % operator

Current behavior: doesn't exist

Desired behavior: % works

Use case:
I have a data set that has a state value of 0: off, 1: heat, 2: cool.
I'd like to convert this value to be cool < off < heat to make the singlestat grafana plugin able to display it nicely. the % operator could be used to reformat the data nicely in this case.

### PR DESCRIPTION
**Pull request by jsternberg**
Fixes #7553.

### CODE REVIEW

File: influxql/ast_test.go [hunk 1]
```diff
@@ -1086,6 +1086,8 @@ func TestReduce(t *testing.T) {
 		{in: `foo(bar(2 + 3), 4)`, out: `foo(bar(5), 4)`},
 		{in: `4 / 0`, out: `0.000`},
 		{in: `1 / 2`, out: `0.500`},
+		{in: `2 % 3`, out: `2`},
+		{in: `5 % 2`, out: `1`},
```
Comment (corylanou) [inline influxql/ast_test.go]: Can you test the zero edge cases and negative edge cases?

Comment (jsternberg) [inline influxql/ast_test.go]: Done. Looks like a had a bug anyway when doing `float % integer` which is now fixed.

**Comment by jwilder** [comment]
@jsternberg @rkuchan Are there docs that need to be updated?

**Comment by rkuchan** [comment]
@jwilder - thanks! I created a docs issue. Does this operator work in both the `SELECT` clause and the `WHERE` clause?

**Comment by jsternberg** [comment]
It should work in the `WHERE` clause, but it isn't. It looks like I missed some stuff. I'll have a follow-up PR to fix that.

**Comment by jsternberg** [comment]
Fixed those issues in #7990.

**Pull request merged**